pkg/internal/codegen/parse/util.go

```diff
@@ -395,3 +425,45 @@ func parseScaleParams(t *types.Type) (map[string]string, error) {
 	}
 	return nil, fmt.Errorf(jsonPathError)
 }
+
+// ready subresource requires name,type,JSONPath fields and rest of the field are optional
+// +kubebuilder:printcolumn:ready:name=<name>,type=<type>,description=<desc>,JSONPath:<.spec.Name>,priority=<0|1>,format=<format>
+func parsePrintColumnParams(t *types.Type) v1beta1.CustomResourceColumnDefinition {
+	config := v1beta1.CustomResourceColumnDefinition{}
+	for _, comment := range t.CommentLines {
+		if strings.Contains(comment, "+kubebuilder:printcolumn:ready") {
+			parts := strings.Replace(comment, "+kubebuilder:printcolumn:ready:", "", -1)
+			part := strings.Split(parts, ",")
+			if len(part) < 3 {
+				fmt.Println(fmt.Errorf("Expected required fields in this context: %s", comment))
+			}
```

Comment:
I think we have parsing key-value pairs helpers under `utils` pkg somewhere. See if we can use those ?

Reply:
yeah, I've seen that function here https://github.com/kubernetes-sigs/controller-tools/blob/master/pkg/generate/internal/util.go#L50 However, I can not reuse it because the package is named `internal`. It's a golang feature to provide encapsulation of the package internals. One workaround here is to copy the same function as is. Let me know your thoughts on this information.

Reply:
I can't reuse  `ParseKV` function listed under `pkg/generate/internal` since there are a few test cases in this feature which are not handled in the existing KV function. Ultimately, I need to create a separate function to handle this logic.